server/src/test/kotlin/org/itmodreamteam/myrest/server/service/restaurant/ReservationTest.kt

```diff
@@ -0,0 +1,148 @@
+@DataJpaTest
+@ContextConfiguration(classes = [ReservationTest.Config::class])
+class ReservationTest {
+
+    @Autowired
+    lateinit var reservationService: ReservationService
+
+    @Autowired
+    lateinit var reservationRepository: ReservationRepository
+
+    @Autowired
+    lateinit var userRepository: UserRepository
+
+    @Autowired
+    lateinit var restaurantRepository: RestaurantRepository
+
+    @Autowired
+    lateinit var restaurantTableRepository: RestaurantTableRepository
+
+    @Autowired
+    lateinit var employeeRepository: EmployeeRepository
+
+    @MockBean
+    lateinit var notificationService: NotificationService
+
+    @MockBean
+    lateinit var currentUserService: CurrentUserService
+
+    private lateinit var reservation: Reservation
+    private lateinit var table: RestaurantTable
+    private lateinit var user1: User
+    private lateinit var user2: User
+    private lateinit var user3: User
+    private lateinit var manager: Manager
+
+    @Before
+    fun setup() {
+        val restaurant = restaurantRepository.save(Restaurant("name", "desc", "legal"))
+        table = restaurantTableRepository.save(RestaurantTable(restaurant, 1, null, 1))
+        user1 = userRepository.save(User("Alex", "Bowie"))
+        user2 = userRepository.save(User("Sam", "Ivanov"))
+        user3 = userRepository.save(User("Karl", "Marks"))
+        manager = employeeRepository.save(Manager(restaurant, user3))
+        val now = LocalDateTime.now()
+        authorize(user1)
+        reservation = reservationRepository.save(Reservation(user1, table, now.plusHours(1), now.plusHours(5)))
+    }
+
+    private fun authorize(user: User) {
+        Mockito.`when`(currentUserService.currentUserEntity).thenReturn(user)
+    }
+
+    @Test
+    fun `When create new reservation, then new reservation created`() {
+        val reservation = reservationRepository.findByIdOrNull(reservation.id)
+        Assertions.assertThat(reservation).isNotNull
+        Assertions.assertThat(reservation?.status).isEqualTo(ReservationStatus.PENDING)
+        Assertions.assertThat(reservation?.user).isEqualTo(user1)
+        Assertions.assertThat(reservation?.table).isEqualTo(table)
+    }
+
+    @Test
+    fun `When create new reservation right after existing one, then new a reservation created`() {
+        authorize(user2)
+        val reservation = reservationService.submitReservationForApproval(
+            table,
+            reservation.activeUntil,
+            reservation.activeUntil.plusHours(2)
+        )
+        Assertions.assertThat(reservation).isNotNull
+        Assertions.assertThat(reservation.status).isEqualTo(ReservationStatus.PENDING)
+        Assertions.assertThat(reservation.user).isEqualTo(user2)
+        Assertions.assertThat(reservation.table).isEqualTo(table)
+    }
+
+    @Test(expected = UserException::class)
+    fun `Given existing reservation, when create reservation to same table and overlapping time, then failure`() {
```

Comment:
when reserve (create reservation) the same table for overlapping time (interval)